src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -244,29 +243,6 @@ private Guardrails()
         MBeanWrapper.instance.registerMBean(this, MBEAN_NAME);
     }
 
-    /**
-     * Whether guardrails are enabled.
-     *
-     * @return {@code true} if guardrails are enabled and daemon is initialized,
-     * {@code false} otherwise (in which case no guardrail will trigger).
-     */
-    public static boolean enabled(ClientState state)
-    {
-        return DatabaseDescriptor.isDaemonInitialized() && CONFIG_PROVIDER.getOrCreate(state).getEnabled();
```

Comment:
We still need the call to `DatabaseDescriptor.isDaemonInitialized` so guardrails are not checked during startup nor by standalone tools like `scrub`. This call should be part of `Guardrail#enabled(ClientState)`.